### ISSUE TITLE
Migrate ``pyreverse`` to ``argparse``

### ISSUE THREAD

**Issue by DanielNoord** [PyCQA/pylint#6279]
### Current problem

Opening a separate issue to track progress here.

### Desired solution

After #6209 has landed I have a PR ready that works on moving `pyreverse` to `argparse`.

After that transition we can add a number of deprecation warnings on unused classes. For example, `OptionsProviderMixIn`.

**Comment by DanielNoord** [comment]
@DudeNr33 Could you help me with one question. How is the ``--class`` option supposed to work?

It has `action` `append`, but I'm not sure how to use it? Reading the `argparse`/`optparse` documentation I should be able to do ``pyreverse --class=foo --class=bar test.py`` but that leads to a crash.
Similar for `pyreverse --class=foo,bar test.py`.

Is that option broken? Or should it be used differently?

**Comment by DudeNr33** [comment]
Hm, on my current local branch (functional-tests-enable-disable), I can run ``pyreverse -c pylint.checkers.base_checker.BaseChecker -c pylint.checkers.base_checker.BaseTokenChecker -o png -d .notes pylint`` just fine. 

I created a dummy file with two classes and experienced an ``InferenceError`` when putting a ``=`` between ``-c`` and the class name:

```python
# file a.py
class Dummy:
    a = 1

class Foo:
    b = 1
```

```console
% pyreverse -c=Dummy -c=Foo a.py
parsing a.py...
Traceback (most recent call last):
  File "/Users/andreas/.pyenv/versions/pylint3102/bin/pyreverse", line 33, in <module>
    sys.exit(load_entry_point('pylint', 'console_scripts', 'pyreverse')())
  File "/Users/andreas/programming/forks/pylint/pylint/__init__.py", line 44, in run_pyreverse
    PyreverseRun(argv or sys.argv[1:])
  File "/Users/andreas/programming/forks/pylint/pylint/pyreverse/main.py", line 214, in __init__
    sys.exit(self.run(args))
  File "/Users/andreas/programming/forks/pylint/pylint/pyreverse/main.py", line 229, in run
    diadefs = handler.get_diadefs(project, linker)
  File "/Users/andreas/programming/forks/pylint/pylint/pyreverse/diadefslib.py", line 217, in get_diadefs
    diagrams.append(generator.class_diagram(project, klass))
  File "/Users/andreas/programming/forks/pylint/pylint/pyreverse/diadefslib.py", line 185, in class_diagram
    klass = next(module.ilookup(klass))
  File "/Users/andreas/.pyenv/versions/3.10.2/envs/pylint3102/lib/python3.10/site-packages/astroid/bases.py", line 143, in _infer_stmts
    raise InferenceError(
astroid.exceptions.InferenceError: Inference failed for all members of ().
```

Don't know where that is coming from, though.

**Comment by DanielNoord** [comment]
> I created a dummy file with two classes and experienced an `InferenceError` when putting a `=` between `-c` and the class name:
> Don't know where that is coming from, though.

The crash is perhaps related to #3307

Thanks for the example! I'll investigate!
